pyflakes/test/harness.py

```diff
@@ -16,8 +16,13 @@ class TestCase(unittest.TestCase):
     withDoctest = False
 
     def flakes(self, input, *expectedOutputs, **kw):
-        tree = compile(textwrap.dedent(input), "<test>", "exec", PyCF_ONLY_AST)
-        w = checker.Checker(tree, withDoctest=self.withDoctest, **kw)
+        if kw.get('scope', None):
+            w = checker.Checker(input.body[0],
```

Comment:
`tree = input.body[0]` then you do not need to modify the `w = checker.Checker(..)` line